libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.processor.ts

```diff
@@ -0,0 +1,175 @@
+import { RuleDataProcessor } from '@carrot-fndn/shared/app/types';
+import { provideDocumentLoaderService } from '@carrot-fndn/shared/document/loader';
+import {
+  isNil,
+  isNonEmptyArray,
+  toDocumentKey,
+} from '@carrot-fndn/shared/helpers';
+import { getEventAttributeValue } from '@carrot-fndn/shared/methodologies/bold/getters';
+import {
+  type DocumentQuery,
+  DocumentQueryService,
+  loadParentDocument,
+} from '@carrot-fndn/shared/methodologies/bold/io-helpers';
+import { DocumentMatcher } from '@carrot-fndn/shared/methodologies/bold/matchers';
+import {
+  eventHasName,
+  eventNameIsAnyOf,
+} from '@carrot-fndn/shared/methodologies/bold/predicates';
+import {
+  type CertificateRewardDistributionOutput,
+  type Document,
+  DocumentEventAttributeName,
+  DocumentEventName,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+import { mapToRuleOutput } from '@carrot-fndn/shared/rule/result';
+import {
+  type RuleInput,
+  type RuleOutput,
+  RuleOutputStatus,
+} from '@carrot-fndn/shared/rule/types';
+import { type NonZeroPositive } from '@carrot-fndn/shared/types';
+import BigNumber from 'bignumber.js';
+import { is } from 'typia';
+
+import type { ResultContentsWithMassIdCertificateValue } from './rewards-distribution.types';
+
+import { RewardsDistributionProcessorErrors } from './rewards-distribution.errors';
+import { calculateRewardsDistribution } from './rewards-distribution.helpers';
+
+const { RULES_METADATA } = DocumentEventName;
+const { REWARDS_DISTRIBUTION_RULE_RESULT_CONTENT, UNIT_PRICE } =
+  DocumentEventAttributeName;
+
+export const RESULT_COMMENTS = {} as const;
+
+interface RuleSubject {
+  creditsDocumentUnitPrice: NonZeroPositive;
+  resultContentsWithMassIdCertificateValue: ResultContentsWithMassIdCertificateValue[];
+}
```

Comment:
I think you are redefining this in a function parameter in the helpers. I think it is better to extract it to a types file